sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DisposeFromDispose.cs

```diff
@@ -84,17 +92,32 @@ private static bool IsDisposeMethodCalled(InvocationExpressionSyntax invocation,
                 methodSymbol.Equals(methodSymbol.ContainingType.FindImplementationForInterfaceMember(disposeMethod));
         }
 
-        private static bool IsDisposableField(ExpressionSyntax expression, SemanticModel semanticModel)
```

Comment:
this method was previously verifying both that it's a disposable field and that the containing type implemented IDisposable, so I split it